core-client/src/lib.rs

```diff
@@ -1581,6 +1590,150 @@ async fn do_restore_from_backup(
     Ok(())
 }
 
+#[allow(clippy::too_many_arguments)]
+pub async fn do_reshare(
+    internal_client: &mut Client,
+    core_endpoints: &HashMap<u32, CoreServiceEndpointClient<Channel>>,
+    rng: &mut AesRng,
+    cmd_conf: &CmdConfig,
+    cc_conf: &CoreClientConfig,
+    destination_prefix: &Path,
+    kms_addrs: &[alloy_primitives::Address],
+    num_parties: usize,
+    param: FheParameter,
+    key_id: RequestId,
+    preproc_id: RequestId,
+    max_retries: usize,
+) -> anyhow::Result<RequestId> {
+    let request_id = RequestId::new_random(rng);
+    // Create the request
+    let request = internal_client.reshare_request(
+        &request_id,
+        &key_id,
+        &preproc_id,
+        Some(param),
+        &dummy_domain(),
+    )?;
+
+    // Send the request
+    let mut req_tasks = JoinSet::new();
+    for (party_id, ce) in core_endpoints.iter() {
+        let req_cloned = request.clone();
+        let mut cur_client = ce.clone();
+        let party_id = *party_id;
+        req_tasks.spawn(async move {
+            (
+                party_id,
+                cur_client
+                    .initiate_resharing(tonic::Request::new(req_cloned))
+                    .await,
+            )
+        });
+    }
+
+    let mut results = Vec::new();
+    while let Some(inner) = req_tasks.join_next().await {
+        let (party_id, result) = inner?;
+        let result = result?.into_inner();
+        assert_eq!(result.request_id, Some(request_id.into()));
+        results.push((party_id, result));
+    }
+
+    if cmd_conf.expect_all_responses {
+        assert_eq!(results.len(), num_parties); // check that we have responses from all parties
+    }
+
+    // Poll the result endpoint
+
+    let mut response_tasks = JoinSet::new();
+    for (party_id, ce) in core_endpoints.iter() {
+        let mut cur_client = ce.clone();
+
+        let party_id = *party_id;
+        response_tasks.spawn(async move {
+            let response_request: tonic::Request<kms_grpc::kms::v1::RequestId> =
+                tonic::Request::new(request_id.into());
+            tokio::time::sleep(tokio::time::Duration::from_millis(
+                SLEEP_TIME_BETWEEN_REQUESTS_MS,
+            ))
+            .await;
+            let mut response = cur_client.get_resharing_result(response_request).await;
+
+            let mut ctr = 0_usize;
+            while response.is_err()
+                && response.as_ref().unwrap_err().code() == tonic::Code::Unavailable
+            {
+                tokio::time::sleep(tokio::time::Duration::from_millis(
+                    SLEEP_TIME_BETWEEN_REQUESTS_MS,
+                ))
+                .await;
+                let response_request: tonic::Request<kms_grpc::kms::v1::RequestId> =
+                    tonic::Request::new(request_id.into());
+                response = cur_client.get_resharing_result(response_request).await;
+                ctr += 1;
+                if ctr >= max_retries {
```

Comment:
I think this should maybe be `cmd_conf.max_iter;`? 
I thought `max_retries` would be the amount of times we try to execute the command until success?